FlashLFQ/FlashLFQResults.cs

```diff
@@ -299,6 +300,181 @@ public void CalculateProteinResultsTop3(bool useSharedPeptides)
+                            {
+                                double sampleIntensity = 0;
+                                double highestFractionIntensity = 0;
+
+                                foreach (var fraction in sample.GroupBy(p => p.Fraction))
+                                {
+                                    double fractionIntensity = 0;
+                                    int replicatesWithValidValues = 0;
+
+                                    foreach (SpectraFileInfo replicate in fraction.OrderBy(p => p.Replicate))
+                                    {
+                                        double replicateIntensity = peptide.GetIntensity(replicate);
+
+                                        if (replicateIntensity > 0)
+                                        {
+                                            fractionIntensity += replicateIntensity;
+                                            replicatesWithValidValues++;
+                                        }
+                                    }
+
+                                    if (replicatesWithValidValues > 0)
+                                    {
+                                        fractionIntensity /= replicatesWithValidValues;
+                                    }
+
+                                    if (fractionIntensity > highestFractionIntensity)
+                                    {
+                                        highestFractionIntensity = fractionIntensity;
+                                        sampleIntensity = highestFractionIntensity;
+                                    }
+                                }
+
+                                int sampleNumber = sample.Key;
+                                sampleIntensity = Math.Log(sampleIntensity, 2);
+
+                                if (double.IsInfinity(sampleIntensity))
+                                {
+                                    sampleIntensity = double.NaN;
+                                }
+
+                                peptideIntensityMatrix[peptidesForThisProtein.IndexOf(peptide), sampleN] = sampleIntensity;
+                            }
+
+                            sampleN++;
+                        }
+                    }
+
+                    // do median polish protein quantification
+                    MedianPolish(peptideIntensityMatrix, 10, 0.0001, out var rowEffects, out var columnEffects, out var overallEffect);
+
+                    // set the sample protein intensities
+                    sampleN = 0;
+                    foreach (var group in SpectraFiles.GroupBy(p => p.SampleGroup).OrderBy(p => p.Key))
+                    {
+                        foreach (var sample in group.GroupBy(p => p.Sample).OrderBy(p => p.Key))
+                        {
+                            double sampleProteinIntensity = Math.Pow(2, columnEffects[sampleN]) * referenceProteinIntensity;
```

Comment:
why square the column effect? i haven't seen that in examples but maybe it makes mathematical sense.